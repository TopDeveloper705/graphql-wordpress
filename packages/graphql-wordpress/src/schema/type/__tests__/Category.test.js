import CategoryType from 'type/Category';
import getLoaders from 'data/loaders';

const args = {};
const context = {};
const info = {
  rootValue: {
    loaders: getLoaders(),
  },
};

// eslint-disable-next-line no-underscore-dangle
const fields = CategoryType._typeConfig.fields();

describe('Test Category type', () => {
  test('Test name', () => {
    expect(CategoryType.name).toMatchSnapshot();
  });

  test('Test type', () => {
    expect(CategoryType.isTypeOf({ taxonomy: 'category' })).toBe(true);
  });

  test('Test fields', () => {
    expect(Object.keys(CategoryType.getFields())).toMatchSnapshot();
  });

  test('Test resolve parent', async () => {
    const parent = await fields.parent.resolve({ parent: 69 }, args, context, info);
    expect(parent).toMatchSnapshot();
  });

  test('Test resolve parent null', () => {
    expect(fields.parent.resolve({ parent: 0 }, args, context, info)).toBeNull();
  });
});
