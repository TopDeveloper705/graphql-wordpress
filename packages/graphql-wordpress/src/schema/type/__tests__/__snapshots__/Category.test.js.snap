// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Category type Test fields 1`] = `
Array [
  "id",
  "description",
  "link",
  "name",
  "slug",
  "taxonomy",
  "count",
  "meta",
  "parent",
]
`;

exports[`Test Category type Test name 1`] = `"Category"`;

exports[`Test Category type Test resolve parent 1`] = `
Category {
  "id": 69,
}
`;
